content/articles/2019-05-vroom-1-0-0.Rmarkdown

```diff
@@ -0,0 +1,450 @@
+files
+data <- vroom(files)
+```
+
+## Reading and writing compressed files
+
+Just like readr, vroom automatically reads and writes zip, gzip, bz2 and xz compressed
+files with the standard file extensions.
+
+```{r}
+vroom_write(flights, "flights.tsv.gz")
+
+# Check file sizes to show file is compressed
+fs::file_size(c("flights.tsv", "flights.tsv.gz"))
+
+# Read the file back in
+data <- vroom("flights.tsv.gz")
+```
+
+# Reading remote files
+
+vroom can also read files from the internet as well by passing the URL of the file to `vroom()`.
+
+```{r}
+file <- "https://raw.githubusercontent.com/r-lib/vroom/master/inst/extdata/mtcars.csv"
+data <- vroom(file)
+```
+
+It can even read gzipped files from the internet (although currently not the other compressed formats).
+
+## Reading and writing from pipe connections
+
+vroom provides efficient input and output from `pipe()` connections.
+
+This is useful for doing things like pre-filtering large inputs with command line tools like [grep](https://en.wikipedia.org/wiki/Grep).
+
+```{r}
+# Return only flights on United Airlines
+data <- vroom(pipe("grep -w UA flights.tsv"), col_names = names(flights))
+```
+
+Or using multi-threaded compression programs like
+[pigz](https://zlib.net/pigz/), which can greatly reduce the time to write compressed
+files.
+
+```{r}
+bench::workout({
+  vroom_write(flights, "flights.tsv.gz")
+  vroom_write(flights, pipe("pigz > flights.tsv.gz"))
+})
+```
+
+## Column selection
+
+`vroom` introduces a new argument `col_select` which makes selecting columns to
+keep (or omit) more straightforward.
+
+The parameter uses the same interface as `dplyr::select()`, so you can do flexible selection operations.
+
+- Select with the column names
+```{r}
+data <- vroom("flights.tsv", col_select = c(year, flight, tailnum))
+```
+- Drop columns by name
+```{r}
+data <- vroom("flights.tsv", col_select = c(-dep_time, -air_time:-time_hour))
+```
+
+- Use numeric indices
+```{r}
+data <- vroom("flights.tsv", col_select = c(1, 3, 11))
+```
+- Use the selection helpers
+```{r}
+data <- vroom("flights.tsv", col_select = ends_with("time"))
+```
+- Or rename columns
+```{r}
+data <- vroom("flights.tsv", col_select = list(plane = tailnum, everything()))
+data
+```
+
+## Name repair
+
+Often the names of columns in the original dataset are not ideal to work with.
+`vroom()` uses the same [.name_repair](https://www.tidyverse.org/articles/2019/01/tibble-2.0.1/#name-repair)
+argument as tibble, so you can use one of the default name repair strategies or
+provide a custom function. A great approach is to use the
+[janitor](http://sfirke.github.io/janitor/) `make_clean_names()` function as the input.
+
+```{r}
+vroom("flights.tsv", .name_repair = janitor::make_clean_names)
+
+vroom("flights.tsv", .name_repair = ~ janitor::make_clean_names(., case = "lower_camel"))
+```
+
+
+## Column types
+
+Like readr, vroom guesses the data types of columns as they are read, however sometimes it
```

Comment:
Mention improved heuristic (i.e. looks at data throughout file, not just first `n` rows)